src/shared/TestInfrastructure/Objects/NullTrace.cs

```diff
@@ -78,6 +78,12 @@ public class NullTrace2 : ITrace2
             string filePath = "",
             int lineNumber = 0) { }
 
+        public void WriteError(string errorMessage,
+            string parameterizedMessage = null,
+            string filePath = "",
```

Comment:
question: in the ITrace2 definition, you add the `...CompilerServices...` thing to (I presume) override the default value to the call-site's filename.  You also use it in the main implementation's declaration.  Do we need that here or do we inherit the default value override from the interface declaration?

(not a criticism, just asking)

Reply:
It's inherited! Good eye though.